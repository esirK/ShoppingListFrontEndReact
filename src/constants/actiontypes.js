export const SUBMITTING_STARTED = 'SUBMITTING_STARTED';
export const GETTING_SHOPPINGLISTS_STARTED = 'GETTING_SHOPPINGLISTS_STARTED';
export const SUBMISSION_SUCCESSFUL = 'SUBMISSION_SUCCESSFUL';
export const SUBMISSION_ERROR = 'SUBMISSION_ERROR';
export const CLEAR_ERRORS = 'CLEAR_ERRORS';
export const SWITCH_USER_AUTH_STATUS = 'SWITCH_USER_AUTH_STATUS';
export const ERROR_ENCOUNTERED = 'ERROR_ENCOUNTERED';
export const SHOPPINGLISTS_LOADED_SUCCESSFULLY = 'SHOPPINGLISTS_LOADED_SUCCESSFULLY';
export const CREATING_SHOPPINGLISTS_STARTED = 'CREATING_SHOPPINGLISTS_STARTED';
export const SHOPPINGLISTS_CREATED_SUCCESSFULY = 'SHOPPINGLISTS_CREATED_SUCCESSFULY';
export const SHOPPINGLISTS_DELETED_SUCCESSFULY = 'SHOPPINGLISTS_DELETED_SUCCESSFULY';
export const SHOPPINGLISTS_UPDATED_SUCCESSFULY = 'SHOPPINGLISTS_UPDATED_SUCCESSFULY';