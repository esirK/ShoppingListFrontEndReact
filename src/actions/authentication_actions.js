import axios from 'axios';
import jwt from 'jsonwebtoken';

import * as types from '../constants/actiontypes';

import {URL, startSubmitting} from './index';

export function submissionSuccessful(message){
	return {
		type: types.SUBMISSION_SUCCESSFUL,
		message
	};
}
export function submissionFailed(error) {
	return{
		type: types.SUBMISSION_ERROR,
		error
	};
}
export function setAuthStatusOfUser(user){
	return {
		type: types.SWITCH_USER_AUTH_STATUS,
		user
	};
}
export function submitDetails(details, callback, route) {
	//route represents the endpoint to send this details
	return (dispatch) => {
		dispatch(startSubmitting());
		return axios.post(`${URL}${route}`, {
			'email': details['email'],
			'name': details['username'],
			'password': details['password']
		}
		).then((data) => {
			if (data.data.token !== undefined){
				//Response has token its a login i.e save the token
				localStorage.setItem('jwt',data.data.token);
				dispatch(setAuthStatusOfUser(jwt.decode(data.data.token)));
			}
			callback();
			dispatch(submissionSuccessful(data.data.message));
		}).catch((error) => {
			if(error.response != null) {
				return dispatch(submissionFailed(error.response.data.message));
			}
			if(!error.status){
				//network error
				return dispatch(submissionFailed(error.message));
			}
		});

	};
}
function userLoggedOut(message){
	return{
		type: types.SHOW_SNACK_BAR,
		message
	};
}
export function Logout(){
	return(dispatch)=>{
		localStorage.removeItem('jwt');
		dispatch(setAuthStatusOfUser({}));
		dispatch(userLoggedOut('You have Been Logged out.'));
	};
}