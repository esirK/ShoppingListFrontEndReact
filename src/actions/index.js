import axios from 'axios';

import * as types from '../constants/actiontypes';

export * from './shoppinglist_actions';
export * from './authentication_actions';
export * from './shoppinglistitems_actions';

export const URL = 'http://127.0.0.1:5000/v1/';


export function resetErrors() {
	return{
		type: types.CLEAR_ERRORS
	};
}
export function startSubmitting(){
	console.log('Started Submitting....');
	return {
		type: types.SUBMITTING_STARTED
	};
}
export function errorEncountered(error){
	return{
		type: types.ERROR_ENCOUNTERED,
		error
	};
}

export function hideSnackBar(dispatch){
	return(dispatch)=>{
		dispatch({type: types.HIDE_SNACK_BAR});
	};
}
/**
 * ShoppingList items
 */

//View Selected Shoppinglist
export function viewShoppingList(id){
	console.log('Looking for the shoppinglist ', id);
	return (dispatch) =>{
		//clear all errors first
		dispatch(resetErrors());
		dispatch(loadingShoppingListItems());
		return axios({
			method: 'get',
			url: `${URL}${'shoppinglists/'}${id}`,
			auth: {
				username: localStorage.getItem('jwt'),
				password: ''
			}
		}).then((response)=>{
			console.log('Got catcha ', response, response.data);
			dispatch(shoppinglistItemsRecieved(response.data, id));
		}).catch(error=>{
			dispatch(loadingShoppingListItemsFailed(error.response.data.message));
			dispatch(errorEncountered(error.response.data.message));
		});
	};
}
export function loadingShoppingListItems(){
	console.log('Loading items dispatched');
	return {
		type: types.SHOPPINGLIST_ITEMS_LOADING,
	};
}
function loadingShoppingListItemsFailed(error){
	return {
		type: types.SHOPPINGLIST_ITEMS_FAILED,
		error
	};
}
export function shoppinglistItemsRecieved(data, id){
	return {
		type: types.SHOPPINGLIST_ITEMS_LOADED_SUCCESSFULLY,
		data, id
	};
}