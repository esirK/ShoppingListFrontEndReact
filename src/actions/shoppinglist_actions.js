import axios from 'axios';


import * as types from '../constants/actiontypes';
import {errorEncountered, URL, resetErrors, startSubmitting} from './index';
/**
 * ShoppingLists
 */

export function getShoppingLists(){
	return (dispatch) => {
		dispatch(getShoppingListStarted());
		return axios({
			method: 'get',
			url: `${URL}${'shoppinglists'}`,
			auth: {
				username: localStorage.getItem('jwt'),
				password: ''
			}
		})
			.then(function (response) {
				console.log('Shoppinglists recieved as',response);
				dispatch(shoppinglistsRecieved(response.data));
			})
			.catch(function (error) {
				console.log('wahhh', error);
				dispatch(errorEncountered(error.message));
			});

	};
}
//Add a new shopping list function
export function addNewShoppingList(details){
	return (dispatch) =>{
		//clear all errors first
		dispatch(resetErrors());
		//Notify that adding a shopping list action has been dispatched
		dispatch(addingShoppinglistStarted());
		return axios({
			method: 'post',
			url: `${URL}${'shoppinglists'}`,
			data: {
				name: details['name'],
				description: details['description']
			  },
			auth: {
				username: localStorage.getItem('jwt'),
				password: ''
			}
		})
			.then(function (response){
				//Dispatch shoppinglist created successfully
				console.log('Got response when adding', response.data);
				dispatch(shoppinglistCreated(response.data.data, response.data.message));
			})
			.catch(function (error){
				//Dispatch shoppinglist creation failed
				console.log(error);
				if(error.message){
					dispatch(errorEncountered(error.message));
				}
				if(error.response.data.message !== 'undefined'){
					console.log('Got errorx', error );
				    dispatch(errorEncountered(error.response.data.message));
				}
			});
	};
}
//Delete an existing shoppinglist function
export function deleteShoppingList(id){
	return (dispatch)=>{
		//clear all errors first
		dispatch(resetErrors());
		dispatch(startSubmitting);
		return axios({
			method: 'delete',
			url: `${URL}${'shoppinglists/'}${id}`,
			auth: {
				username: localStorage.getItem('jwt'),
				password: ''
			}
		}).then((response)=>{
			//Dispatch shoppinglist DELETION Successfully
			dispatch(shoppinglistDeleted(response.data.message, response.data.data));
			console.log('Got response', response.data.message);
		}).catch((error)=>{
			//Dispatch shoppinglist DELETION Failed
			dispatch(errorEncountered(error.message));
		});
	};
}

export function updateShoppingList(id, details){
	//Updates an existing shoppinglist
	return (dispatch) =>{
		//clear all errors first
		dispatch(resetErrors());
		dispatch(updatingShoppinglistStarted());
		return axios({
			method: 'put',
			url: `${URL}${'shoppinglists/'}${id}`,
			data: {
				new_name: details['new_name'],
				description: details['description']
			  },
			auth: {
				username: localStorage.getItem('jwt'),
				password: ''
			}
		}).then((response)=>{
			if((response.data.message).includes('Nothing was provided')){
				dispatch(errorEncountered(response.data.message));
			}else{
				dispatch(closeUpdateDialog());
				dispatch(shoppinglistUpdated(response.data.message, response.data.data));
			}
		}).catch((error)=>{
			console.log('Got Meso', error);
			dispatch(errorEncountered(error.message));
		});
	};
}
function getShoppingListStarted(){
	console.log('Getting started');
	return{
		type: types.GETTING_SHOPPINGLISTS_STARTED
	};
}
function shoppinglistsRecieved(response){
	return{
		type: types.SHOPPINGLISTS_LOADED_SUCCESSFULLY,
		response
	};
}

function addingShoppinglistStarted(){
	return {
		type: types.CREATING_SHOPPINGLISTS_STARTED,
	};
}

function shoppinglistCreated(response, message){
	return {
		type: types.SHOPPINGLISTS_CREATED_SUCCESSFULY,
		response, message
	};
}
function shoppinglistDeleted(message, response){
	return {
		type: types.SHOPPINGLISTS_DELETED_SUCCESSFULY,
		message, response
	};
}
function updatingShoppinglistStarted(){
	return {
		type: types.UPDATING_SHOPPINGLIST_STARTED,
	};
}
function shoppinglistUpdated(message, response){
	return {
		type: types.SHOPPINGLISTS_UPDATED_SUCCESSFULY,
		message, response
	};
}
export function activateFab(){
	return(dispatch)=>{
		dispatch(
		 {type: types.ACTIVATE_FAB}
		);
	};
}
export function closeFab(){
	return(dispatch)=>{
		dispatch(
		 {type: types.DEACTIVATE_FAB}
		);
	};
}
export function openUpdateDialog(){
	return(dispatch)=>{
		dispatch({type: types.OPEN_UPDATE_DIALOG});
	};
}
export function closeUpdateDialog(){
	resetErrors();
	return(dispatch)=>{
		dispatch(
			{type: types.CLOSE_UPDATE_DIALOG}
		   );
	};
}