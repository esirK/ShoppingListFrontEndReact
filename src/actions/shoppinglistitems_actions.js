import axios from 'axios';

import * as types from '../constants/actiontypes';

import {errorEncountered, URL, resetErrors} from './index';

export function activateAddItem(){
	//Sets state of add item dialog to true
	return(dispatch)=>{
		dispatch(
			{type: types.ACTIVATE_ADD_ITEM}
		);
	};
}
export function deactivateAddItem(){
	//Sets state of add item dialog to false
	return(dispatch)=>{
		dispatch(resetErrors());
		dispatch(
			{type: types.DEACTIVATE_ADD_ITEM}
		);
	};
}
export function addNewShoppingListItem(details){
	return (dispatch) => {
		console.log('see whst you doing ', details);
		//clear all errors first
		dispatch(resetErrors());
		dispatch(addingShoppinglistItemStarted());
		return axios({
			method: 'post',
			url: `${URL}${'shoppinglist_items'}`,
			data:{
				name: details['name'].trim(),
				price: details['price'].trim(),
				quantity: details['quantity'].trim(),
				shoppinglist_id: details['shoppinglist_id']
			},
			auth: {
				username: localStorage.getItem('jwt'),
				password: ''
			}
		})
			.then(function (response) {
				console.log('Here is response of adding a shoppinglist item ', response.data);
				dispatch(shoppinglistItemCreated(response.data.message));
			})
			.catch(function (error) {
				console.log('wahhh', error.response);
				if(error.response.data.error === undefined){
					dispatch(errorEncountered(error.response.data.message));
				}else{
					dispatch(errorEncountered(error.response.data.error));
				}
			});

	};
}
export function deleteShoppinglistItem(id){
	return(dispatch)=>{
		dispatch(deletingItemStarted());
		return axios(
			{
				method: 'delete',
				url: `${URL}${'shoppinglist_items/'}${id}`,
				auth: {
					username: localStorage.getItem('jwt'),
					password: ''
				}
			}
		).then(function(response)
		{
			console.log('Got this response on deleting an item', response);
			dispatch(itemDeletedSuccessfully(response.data.message));
		}
		).catch(function(error){
			console.log('Error while trying to delete ', error);
		});
	};
}
function addingShoppinglistItemStarted(){
	return{
		type: types.ADDING_SHOPPINGLIST_ITEM_STARTED
	};
}
function shoppinglistItemCreated(message){
	return{
		type: types.SHOPPINGLIST_ITEM_ADDED, message
	};
}
function deletingItemStarted(){
	return{
		type: types.DELETING_ITEM_STARTED,
	};
}
function itemDeletedSuccessfully(message){
	return{
		type: types.ITEM_DELETED_SUCCESSFULLY,
		message
	};
}