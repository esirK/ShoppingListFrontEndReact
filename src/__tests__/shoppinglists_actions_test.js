import React from 'react';
import thunk from 'redux-thunk';

import * as types from '../constants/actiontypes';
import * as actions from '../actions';

import configureMockStore from 'redux-mock-store';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


const store = mockStore();

global.window = {};
import localStorage from 'mock-local-storage';
window.localStorage = global.localStorage;

describe('shoppinglists actions', ()=>{
	beforeEach(()=>{
		//Login this User before each test
		return store.dispatch(actions.submitDetails({
			'email': 'kimani@andela.com',
			'password': 'Andela1'
		},() =>{},'user')).then(()=>{
		});
	});
	it('Returns all Shoppinglists when all parameter in getShoppingLists is set to true',()=>{
		store.clearActions();//Remove the login actions
		return store.dispatch(
			actions.getShoppingLists(1,4,true)).then(()=>{
			expect(store.getActions()).toMatchSnapshot();
		});
	});
	it('Returns only specified number in limit of Shoppinglists when all parameter in getShoppingLists is set to false',()=>{
		store.clearActions();//Remove the login actions
		return store.dispatch(
			actions.getShoppingLists(1,4,false)).then(()=>{
			expect(store.getActions()).toMatchSnapshot();
		});
	});
	it('Returns 404 if page specified is not found ', ()=>{
		store.clearActions();//Remove the previous actions
		return store.dispatch(
			actions.getShoppingLists(10,4,false)).then(()=>{
			expect(store.getActions()).toMatchSnapshot();
		});
	});
});