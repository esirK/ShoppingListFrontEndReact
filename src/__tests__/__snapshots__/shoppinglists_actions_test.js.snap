// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoppinglists actions Returns 404 if page specified is not found  1`] = `
Array [
  Object {
    "type": "GETTING_SHOPPINGLISTS_STARTED",
  },
  Object {
    "error": "Request failed with status code 404",
    "type": "ERROR_ENCOUNTERED",
  },
]
`;

exports[`shoppinglists actions Returns all Shoppinglists when all parameter in getShoppingLists is set to true 1`] = `
Array [
  Object {
    "type": "GETTING_SHOPPINGLISTS_STARTED",
  },
  Object {
    "response": Array [
      Object {
        "description": "Beasts exists",
        "id": "166",
        "items": Array [
          Object {
            "id": "32",
            "name": "Monster",
            "price": "200.0",
            "quantity": "2.5",
          },
          Object {
            "id": "39",
            "name": "Jerry Smith",
            "price": "2.0",
            "quantity": "1.0",
          },
        ],
        "name": "Real Beast",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "I am an Evil Morty",
        "id": "78",
        "items": Array [
          Object {
            "id": "21",
            "name": "Jerry Smiths",
            "price": "200.0",
            "quantity": "1.0",
          },
          Object {
            "id": "24",
            "name": "Waga",
            "price": "5.0",
            "quantity": "-4.0",
          },
          Object {
            "id": "19",
            "name": "Sima",
            "price": "20.0",
            "quantity": "4.0",
          },
          Object {
            "id": "25",
            "name": "trison",
            "price": "5.0",
            "quantity": "6.0",
          },
          Object {
            "id": "20",
            "name": "Hahah",
            "price": "2.0",
            "quantity": "3.0",
          },
          Object {
            "id": "18",
            "name": "Uji",
            "price": "25.0",
            "quantity": "5.0",
          },
          Object {
            "id": "22",
            "name": "ugali",
            "price": "2.0",
            "quantity": "3.0",
          },
        ],
        "name": "Evil Morty",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "1000 years of me and mortty",
        "id": "79",
        "items": Array [
          Object {
            "id": "16",
            "name": "Jerry",
            "price": "100.0",
            "quantity": "1.0",
          },
          Object {
            "id": "17",
            "name": "Vans",
            "price": "2500.0",
            "quantity": "1.0",
          },
          Object {
            "id": "29",
            "name": "Beth",
            "price": "2.0",
            "quantity": "3.0",
          },
        ],
        "name": "Rick Sanchez",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "Things to buy when am hedding back to school",
        "id": "168",
        "items": Array [
          Object {
            "id": "56",
            "name": "Transmara",
            "price": "4.0",
            "quantity": "5.0",
          },
          Object {
            "id": "52",
            "name": "Book",
            "price": "10.0",
            "quantity": "2.0",
          },
          Object {
            "id": "49",
            "name": "Biros",
            "price": "20.0",
            "quantity": "5.0",
          },
        ],
        "name": "Back to school",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "",
        "id": "281",
        "items": Array [
          Object {
            "id": "62",
            "name": "umamas",
            "price": "8.0",
            "quantity": "9.0",
          },
          Object {
            "id": "73",
            "name": "Unga",
            "price": "50.5",
            "quantity": "100.5",
          },
          Object {
            "id": "74",
            "name": "sa",
            "price": "32.0",
            "quantity": "323.0",
          },
          Object {
            "id": "76",
            "name": "sdas",
            "price": "32.0",
            "quantity": "322.0",
          },
        ],
        "name": "ujinga",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "dsds",
        "id": "229",
        "items": Array [],
        "name": "Really Hard",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "",
        "id": "253",
        "items": Array [
          Object {
            "id": "59",
            "name": "Steer wheel",
            "price": "400.0",
            "quantity": "2.0",
          },
        ],
        "name": "BMW",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "Buy Tiny rick artirel",
        "id": "336",
        "items": Array [],
        "name": "Tiny Rick",
        "shared": "True",
        "shared_by": "kimani@andela.com",
      },
      Object {
        "description": "Djikista",
        "id": "330",
        "items": Array [
          Object {
            "id": "77",
            "name": "Sababa",
            "price": "9.0",
            "quantity": "6.0",
          },
        ],
        "name": "New",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "Sir God forbid Ufala",
        "id": "233",
        "items": Array [],
        "name": "God forbid",
        "shared": "False",
        "shared_by": "nobody",
      },
    ],
    "type": "SHOPPINGLISTS_LOADED_SUCCESSFULLY",
  },
  Object {
    "response": Array [
      Object {
        "description": "Beasts exists",
        "id": "166",
        "items": Array [
          Object {
            "id": "32",
            "name": "Monster",
            "price": "200.0",
            "quantity": "2.5",
          },
          Object {
            "id": "39",
            "name": "Jerry Smith",
            "price": "2.0",
            "quantity": "1.0",
          },
        ],
        "name": "Real Beast",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "I am an Evil Morty",
        "id": "78",
        "items": Array [
          Object {
            "id": "21",
            "name": "Jerry Smiths",
            "price": "200.0",
            "quantity": "1.0",
          },
          Object {
            "id": "24",
            "name": "Waga",
            "price": "5.0",
            "quantity": "-4.0",
          },
          Object {
            "id": "19",
            "name": "Sima",
            "price": "20.0",
            "quantity": "4.0",
          },
          Object {
            "id": "25",
            "name": "trison",
            "price": "5.0",
            "quantity": "6.0",
          },
          Object {
            "id": "20",
            "name": "Hahah",
            "price": "2.0",
            "quantity": "3.0",
          },
          Object {
            "id": "18",
            "name": "Uji",
            "price": "25.0",
            "quantity": "5.0",
          },
          Object {
            "id": "22",
            "name": "ugali",
            "price": "2.0",
            "quantity": "3.0",
          },
        ],
        "name": "Evil Morty",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "1000 years of me and mortty",
        "id": "79",
        "items": Array [
          Object {
            "id": "16",
            "name": "Jerry",
            "price": "100.0",
            "quantity": "1.0",
          },
          Object {
            "id": "17",
            "name": "Vans",
            "price": "2500.0",
            "quantity": "1.0",
          },
          Object {
            "id": "29",
            "name": "Beth",
            "price": "2.0",
            "quantity": "3.0",
          },
        ],
        "name": "Rick Sanchez",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "Things to buy when am hedding back to school",
        "id": "168",
        "items": Array [
          Object {
            "id": "56",
            "name": "Transmara",
            "price": "4.0",
            "quantity": "5.0",
          },
          Object {
            "id": "52",
            "name": "Book",
            "price": "10.0",
            "quantity": "2.0",
          },
          Object {
            "id": "49",
            "name": "Biros",
            "price": "20.0",
            "quantity": "5.0",
          },
        ],
        "name": "Back to school",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "",
        "id": "281",
        "items": Array [
          Object {
            "id": "62",
            "name": "umamas",
            "price": "8.0",
            "quantity": "9.0",
          },
          Object {
            "id": "73",
            "name": "Unga",
            "price": "50.5",
            "quantity": "100.5",
          },
          Object {
            "id": "74",
            "name": "sa",
            "price": "32.0",
            "quantity": "323.0",
          },
          Object {
            "id": "76",
            "name": "sdas",
            "price": "32.0",
            "quantity": "322.0",
          },
        ],
        "name": "ujinga",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "dsds",
        "id": "229",
        "items": Array [],
        "name": "Really Hard",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "",
        "id": "253",
        "items": Array [
          Object {
            "id": "59",
            "name": "Steer wheel",
            "price": "400.0",
            "quantity": "2.0",
          },
        ],
        "name": "BMW",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "Buy Tiny rick artirel",
        "id": "336",
        "items": Array [],
        "name": "Tiny Rick",
        "shared": "True",
        "shared_by": "kimani@andela.com",
      },
      Object {
        "description": "Djikista",
        "id": "330",
        "items": Array [
          Object {
            "id": "77",
            "name": "Sababa",
            "price": "9.0",
            "quantity": "6.0",
          },
        ],
        "name": "New",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "Sir God forbid Ufala",
        "id": "233",
        "items": Array [],
        "name": "God forbid",
        "shared": "False",
        "shared_by": "nobody",
      },
    ],
    "type": "ALL_SHOPPINGLISTS_LOADED",
  },
]
`;

exports[`shoppinglists actions Returns only specified number in limit of Shoppinglists when all parameter in getShoppingLists is set to false 1`] = `
Array [
  Object {
    "type": "GETTING_SHOPPINGLISTS_STARTED",
  },
  Object {
    "response": Array [
      Object {
        "description": "Beasts exists",
        "id": "166",
        "items": Array [
          Object {
            "id": "32",
            "name": "Monster",
            "price": "200.0",
            "quantity": "2.5",
          },
          Object {
            "id": "39",
            "name": "Jerry Smith",
            "price": "2.0",
            "quantity": "1.0",
          },
        ],
        "name": "Real Beast",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "I am an Evil Morty",
        "id": "78",
        "items": Array [
          Object {
            "id": "21",
            "name": "Jerry Smiths",
            "price": "200.0",
            "quantity": "1.0",
          },
          Object {
            "id": "24",
            "name": "Waga",
            "price": "5.0",
            "quantity": "-4.0",
          },
          Object {
            "id": "19",
            "name": "Sima",
            "price": "20.0",
            "quantity": "4.0",
          },
          Object {
            "id": "25",
            "name": "trison",
            "price": "5.0",
            "quantity": "6.0",
          },
          Object {
            "id": "20",
            "name": "Hahah",
            "price": "2.0",
            "quantity": "3.0",
          },
          Object {
            "id": "18",
            "name": "Uji",
            "price": "25.0",
            "quantity": "5.0",
          },
          Object {
            "id": "22",
            "name": "ugali",
            "price": "2.0",
            "quantity": "3.0",
          },
        ],
        "name": "Evil Morty",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "1000 years of me and mortty",
        "id": "79",
        "items": Array [
          Object {
            "id": "16",
            "name": "Jerry",
            "price": "100.0",
            "quantity": "1.0",
          },
          Object {
            "id": "17",
            "name": "Vans",
            "price": "2500.0",
            "quantity": "1.0",
          },
          Object {
            "id": "29",
            "name": "Beth",
            "price": "2.0",
            "quantity": "3.0",
          },
        ],
        "name": "Rick Sanchez",
        "shared": "False",
        "shared_by": "nobody",
      },
      Object {
        "description": "Things to buy when am hedding back to school",
        "id": "168",
        "items": Array [
          Object {
            "id": "56",
            "name": "Transmara",
            "price": "4.0",
            "quantity": "5.0",
          },
          Object {
            "id": "52",
            "name": "Book",
            "price": "10.0",
            "quantity": "2.0",
          },
          Object {
            "id": "49",
            "name": "Biros",
            "price": "20.0",
            "quantity": "5.0",
          },
        ],
        "name": "Back to school",
        "shared": "False",
        "shared_by": "nobody",
      },
    ],
    "type": "SHOPPINGLISTS_LOADED_SUCCESSFULLY",
  },
]
`;
